### ISSUE TITLE
Merge Async Failure Additions plugin into core package

### ISSUE THREAD

**Issue by jongpie** [jongpie/NebulaLogger#473]
@jamessimone I'd love to know your thoughts on this one - I'd like to merge [your Async Failure Additions plugin](https://github.com/jongpie/NebulaLogger/tree/v4.9.11/nebula-logger/plugins/async-failure-additions) into the core package so that out of the box, Nebula Logger...
- Automatically handles logging all `BatchApexErrorEvent` records
- Automatically handles logging all `FlowExecutionErrorEvent` records
- Provides a ready-to-use implementation for `System.Finalizer`

I think that'd be really convenient for every org to just automatically have that functionality without needing to install an extra package. Any concerns with deprecating the plugin package & moving the metadata into core?

**Comment by jamessimone** [comment]
The only concern I have is if existing implementations already have a `BatchApexErrorEvent` handler - I guess since this is feature-flagged, as is, it should be fine!

**Comment by jongpie** [comment]
Yeah, that concern makes sense, maybe the config should be disabled by default? I'll try to tinker with this idea more in the coming weeks

**Comment by TrangOul** [comment]
@jongpie, in #844 you've mentioned the inconvenience of having separate plugins and that at least for Approval results core package is preferred.
Can we have Async Failure Additions also merged into the core package?

**Comment by jongpie** [comment]
@TrangOul yup, that is the intent of this issue. However, this one is much more complex than the Approval results in issue #844 - for the Approval stuff, it effectively only requires a handful of new methods to provide the functionality, so putting it into core is ideal. 

For the Async Failure Additions, there is a lot more metadata involved at the moment, so putting it into a separate plugin package has been a great way to effectively have it as a production-ready add-on that can much more easily be changed (compared to metadata that's added to the core package). I am also planning to expand it to handle issue #734 so that it provides automatic instrumentation/logging for _all_ Apex errors, using the new free tier of Event Monitoring. I think that will be an incredibly useful feature for a lot of teams/Salesforce orgs, but I want to ensure that the functionality & architecture is "right" & scalable before merging it into the core package.

I don't know for sure when I'll work on this, but ideally, I'd love to have 2 related releases done this year:
- Enhance the existing plugin package to also handle #734 --> have some teams use it & provide feedback.
- Once the plugin package's enhancements have been solidified, then I'll merge it into the core package (and the plugin package will then be considered deprecated).
